app/controllers/products_controller.rb

```diff
@@ -0,0 +1,65 @@
+class ProductsController < ApplicationController
+  before_action :set_product, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @products = Product.all
+  end
+
+  def show
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def edit
+  end
+
+  def create
+    @product = Product.new(product_params)
+
+    respond_to do |format|
+      if @product.save
+        @product.image.attach(params[:product][:image]) if params[:product][:image].present?
+        format.html { redirect_to @product, notice: 'Product was successfully created.' }
+        format.json { render :show, status: :created, location: @product }
```

Comment:
Are you using this as an API? Do you need the format `json`?

I'm pretty sure you can remove it everywhere :)